mushroomsoft/app/src/app/components/contact/contact.component.ts

```diff
@@ -1,51 +1,23 @@
-import {Component} from '@angular/core';
+import {Component, OnInit} from '@angular/core';
 import {FormGroup} from '@angular/forms';
 import {FormlyFormOptions, FormlyFieldConfig} from '@ngx-formly/core';
+import {MianLibService} from '@mushroomsoft-lib';
 
 @Component({
   selector: 'app-contact',
   templateUrl: './contact.component.html',
   styleUrls: ['./contact.component.scss']
 })
-export class ContactComponent {
+export class ContactComponent implements OnInit {
   form = new FormGroup({});
-  model: any = {};
+  model: {} = {};
   options: FormlyFormOptions = {};
-  fields: FormlyFieldConfig[] = [
-    {
-      key: 'Name',
-      type: 'input',
-      templateOptions: {
-        label: 'Name',
-        placeholder: 'Name'
-      }
-    },
-    {
-      key: 'Phone',
-      type: 'input',
-      templateOptions: {
-        label: 'Phone',
-        placeholder: 'Phone number'
-      }
-    },
-    {
-      key: 'Email',
-      type: 'input',
-      templateOptions: {
-        label: 'Email Address',
-        placeholder: 'Email'
-      }
-    },
-    {
-      key: 'Message',
-      type: 'textarea',
-      templateOptions: {
-        label: 'Message',
-        placeholder: 'Message'
-      }
-    }
-  ];
+  fields: FormlyFieldConfig[] = [];
+  constructor(private libService: MianLibService) {}
 
+  ngOnInit(): void {
+    this.fields = this.libService.getContact();
+  }
   submit() {
     alert(JSON.stringify(this.model));
```

Comment:
Don't use alert, instead, you can use a third-party library like to `ngx-toastr`

Reply:
This line that contains the alert is only for testing, because the DNS still needs to be validated to place the code block for sending mail